### PR TITLE
Use psql db in unit tests

### PR DESCRIPTION
**Pull request by croissanne**

### CODE REVIEW

**Review by achilleas-k — DISMISSED** [review]
This is good.  I like this.

**Review by achilleas-k — APPROVED** [review]
Still great.

**Review by kingsleyzissou — DISMISSED** [review]
Oh, very nice!!

File: internal/tutils/psql.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+func containerRuntime() (string, error) {
+	out, err := exec.Command("which", "podman").Output()
+	if err == nil {
+		return strings.TrimSpace(string(out)), nil
+	}
+	out, err = exec.Command("which", "docker").Output()
+	if err == nil {
+		return strings.TrimSpace(string(out)), nil
+	}
+	return "", err
```
Comment (ondrejbudai) [inline internal/tutils/psql.go:35]: nitpick: I would add a message that no container runtime was found and that the user needs to have one.

File: internal/tutils/psql.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		if err != nil {
+			time.Sleep(time.Second * 1)
+			continue
+		}
+		break
+	}
+	return p, nil
+}
+
+func (p *PSQLContainer) execCommand(args []string) (string, error) {
```
Comment (ondrejbudai) [inline internal/tutils/psql.go:87]: nitpick: `args ...string` would make the API similar to the one `exec.Command` has.

File: internal/tutils/psql.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	p := &PSQLContainer{
+		name: name,
+		id:   strings.TrimSpace(string(out)),
+		port: port,
+	}
+
+	tries := 0
+	for tries < 10 {
```
Comment (ondrejbudai) [inline internal/tutils/psql.go:72]: Maybe return an error explicitly if the container failed to start? Might be overcautious, though...

File: internal/tutils/psql.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+)
+
+type PSQLContainer struct {
+	name string
+	id   string
+	port int
+}
+
+const (
+	image string = "postgres:13"
```
Comment (ondrejbudai) [inline internal/tutils/psql.go:22]: `podman` prefers fully-qualified names, so I suggest making this `docker.io/library/postgres:13`.

File: internal/v1/server_test.go [hunk 5]
```diff
@@ -26,7 +26,23 @@ import (
...
+func TestMain(m *testing.M) {
+	d, err := tutils.NewPSQLContainer()
+	if err != nil {
+		panic(err)
+	}
+
+	dbc = d
+	code := m.Run()
+
+	err = dbc.Stop()
```
Comment (ondrejbudai) [inline internal/v1/server_test.go:40]: I wonder whether we can somehow put this into a `defer` block. I would love making this as error-proof as possible, as I don't like the sound of dangling database containers on my system.

**Review by ondrejbudai — COMMENTED** [review]
Lovely! Fewer code paths is always amazing. :)

**Review by ondrejbudai — APPROVED** [review]
Amazing! :)

**Comment by croissanne** [comment]
going to force merge, dependencies on eph are broken :(

**Pull request merged**